tests/sycl/test_barrier.sycl

```diff
@@ -0,0 +1,53 @@
+/**
+ * traccc library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <gtest/gtest.h>
+
+#include <CL/sycl.hpp>
+
+#include "../../device/sycl/src/utils/barrier.hpp"
+
+TEST(CUDABarrier, BarrierAnd) {
+    ::sycl::queue queue;
```

Comment:
```suggestion
    cl::sycl::queue queue;
```

The convention is that if you include `<CL/sycl.hpp>`, you would get the types in the `cl::sycl::` namespace. With `<sycl/sycl.hpp>` you would get them in the `sycl::` one.

Intel allows you to mix these, but for instance [AdaptiveCpp](https://github.com/AdaptiveCpp/AdaptiveCpp) doesn't.

And yes, since `<CL/sycl.hpp>` is an "older standard", I've been using that everywhere so far.